lib/atlas/graph_from_yaml.rb

```diff
@@ -6,21 +6,17 @@ def self.build(graph_yaml)
 
     def initialize(graph_yaml)
       @graph_yaml = graph_yaml
-      @graph      = GraphBuilder.build
+      @graph      = Refinery::Catalyst::FromTurbine.call(GraphBuilder.build)
     end
 
     def build_graph
-      @graph.nodes.each do |node|
-        if node_attributes = @graph_yaml.fetch(:nodes)[node.key]
-          atlas_node        = node.get(:model)
-          atlas_node.input  = node_attributes[:in]
-          atlas_node.output = node_attributes[:out]
-
-          node_attributes.except(:in, :out).each_pair do |attr, val|
-            node.set(attr, val)
-          end
+      graph_nodes.each_pair do |node, attributes|
+        attributes.fetch(:node).each_pair do |attr, val|
+          node.set(attr, val)
+        end
 
-          set_slots(atlas_node, node_attributes.slice(:in, :out))
+        attributes.fetch(:slot).each_pair do |direction, attributes|
```

Comment:
A lot more readable would be:
```
update_slots(node.slots.in, slot_attributes[:in])
update_slots(node.slots.out, slot_attributes[:out])
```

Also, I think using `public_send` is often a code smell :wink: